skrf/multiNoisyNetworkSystem.py

```diff
@@ -0,0 +1,421 @@
+
+        >>> mns.connect('attn1', 2, 'attn2', 1)
+        >>> mns.external_port('attn1', 1, 2)
+        >>> mns.external_port('attn2', 2, 1)
+
+      
+        """
+        if ntwk_name in self.ntwk_dict:
+            self.ntwk_dict[ntwk_name]['connections'][port_num - 1] = ('EXT', external_port_number)
+
+    def verify(self):
+        """
+        Verifies if all the ports within the network have been connected correctly for the `reduce` method.
+
+        Although it is not necessary to call this method, this method will provide the user with any errors
+        they may have made in the connection and port labeling proceedures. 
+
+        The function will return a dictionary of all the errors and warnings that should be addressed before 
+        calling `reduce`.
+
+        Returns
+        -----------
+        output : dictionary
+            A dictionary containing errors and warnings about the verification proceedure
+
+        Examples
+        -----------
+        Before calling `reduce` the method `verify` can be called to determine whether or not any errors have
+        been made in the connection and labeling proceedures.
+
+        >>> import skrf as rf
+        >>> ntwk1 = rf.NoisyNetwork('attenuator1.s2p')
+        >>> ntwk2 = rf.NoisyNetwork('attenuator2.s2p')
+        >>> mns = rf.MultiNoisyNetworkSystem()
+        >>> mns.add(ntwk, 'attn1')
+        >>> mns.add(ntwk, 'attn2')
+        >>> mns.connect('attn1', 2, 'attn2', 1)
+        >>> mns.external_port('attn1', 1, 1)
+        >>> mns.external_port('attn2', 2, 2)
+        >>> er = mns.verify()
+        >>> print(er)
+        """
+
+        errors = {}
+        enumb = 0
+        exts = []
+        lens = []
+        first = True
+        ln = 0
+        same_length = True
+        for ntwk in self.ntwk_dict:
+            if first:
+                first = False
+                ln = len(self.ntwk_dict[ntwk]['ntwk'].frequency)
+            if ln != len(self.ntwk_dict[ntwk]['ntwk'].frequency):
+                enumb += 1
+                errors[enumb] = ('Number of frequencies for each network must be the same'  )
+
+            for p, connection in enumerate(self.ntwk_dict[ntwk]['connections']):
+                if connection == None:
+                    enumb += 1
+                    errors[enumb] = ('Network [' + ntwk + '] port [' + str(p) + '] not connected to anything'  )
+                elif connection[0] == 'EXT':
+                    exts.append(connection)
+
+        num_exts = len(exts)
+        ext_expected_ports = list(range(1, num_exts + 1))
+        for ext in exts:
+            if ext[1] in ext_expected_ports:
+                ext_expected_ports.remove(ext[1])
+            else:
+                enumb += 1
+                errors[enumb] = ('External ports must have correct assignments, e.g. 1, 2, 3, ... ')
+                break
+        
+        return enumb, errors
+
+    def reduce(self):
```

Comment:
I didn't go into the implementation, but isn't it equivalent to cascade multiple NoisyNetwork together :

`reduced_nntwk = nntwk1 ** nntwk2 ** ... ** nntwkN`

?

Reply:
No, the point of multiNoisyNetworkSystem.py is to work with multiport NoisyNetworks. For example, you can combine multiport splitters and three port circulators with this object in any combination. The reduce() function returns a new NoisyNetwork that is not necessarily a two-port. 

This IS very similar to your circuit.py module. The problem is that I don't currently know how to combine noise covariance matrices in a way that works with your circuit.py code. But, one of the things @mgrando needs in his work is the ability to combine multiport noisy networks. So I was kinda stuck and needed a mechanism that would do something similar to circuit.py but work with NoisyNetworks. I'm open to suggestions if you have any ideas on how to perhaps make this a part of circuit.py

Reply:
ok, see my comment below